cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/tester/SlangTestRunner.java

```diff
@@ -252,6 +264,32 @@ private void printTestForActualRunSummary(TestCaseRunMode runMode, Map<String, S
         return resultMap;
     }
 
+    private boolean enabledByBuildMode(
+            SlangBuildMain.BuildMode buildMode,
+            SlangTestCase slangTestCase,
+            Set<String> changedExecutables,
+            Map<String, Executable> allTestedFlowModels) {
+        switch (buildMode) {
+            case BASIC:
+                return true;
+            case CHANGED:
+                return isAffectedTestCase(slangTestCase, changedExecutables, allTestedFlowModels);
+            default:
+                throw new RuntimeException(MessageConstants.NOT_IMPLEMENTED_MESSAGE);
+        }
+    }
+
+    private boolean isAffectedTestCase(SlangTestCase slangTestCase, Set<String> changedExecutables, Map<String, Executable> allTestedFlowModels) {
+        String testFlowPath = slangTestCase.getTestFlowPath();
+        Executable testCaseReference = allTestedFlowModels.get(testFlowPath);
+        if (testCaseReference == null) {
+            throw new RuntimeException("Test case reference[" + testFlowPath + "] not found in compiled models.");
+        }
+        Set<String> testCaseDependencies = new HashSet<>(dependenciesHelper.fetchDependencies(testCaseReference, allTestedFlowModels));
```

Comment:
Maybe I am missing something, but I think from the github interface that this is shallow cloning the fetchDependencies set with another set, so please ignore if this is not the case.